### PR TITLE
[CS2113-F15-2] LifeTrack

### PR DESCRIPTION
**Pull request by paturikarthik**
LifeTrack is a desktop app for students to track their health data, optimized for use via a Command Line Interface (CLI). LifeTrack is especially catered towards those who can type fast, as they can track their health data much faster than when using traditional Graphical User Interface (GUI) apps.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -5,9 +5,146 @@
...
+
+### Calories list feature
+
+The `calories list` feature lists out the record of all the Calories data that the user has keyed in. The Calories data are all stored into a `ArrayList<Entry> calorieArrayList` attribute of the `CalorieList` Class. Calories data are printed when the `printCalorieList()` function is called. 
+
+The `printCalorieList()` function iterates through the `calorieArrayList` and prints out the Entries according to its order in the Array List.
+
+The Class diagram for Calories list feature is shown below. Unrelated attributes and Classes were excluded.
+
+![CaloriesListClassDiagram](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/rexyyong/tp/DevGuideRex/docs/CaloriesListClassDiagram.puml)
```
Comment (cayennegoh) [inline docs/DeveloperGuide.md]: Could use these proper symbols for representing public/private classes instead of squares
+ : public
- : private
# : protected
~ : package private

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -5,9 +5,146 @@
...
+
+- Step 3: The string will be divided to two substrings according to the command syntax. Index will be tried to get from the second substring by `Integer.parseInt()`.
+
+- Step 4: The calories record (`Entry`) stored in the `ArrayList<Entry> caloriesList` will be deleted by calling `calorieArrayList.remove((index-1));` and a successful deleting message will be shown in terminal by calling `CalorieListUi#successfulDeletedMessage(toDelete)`
+
+- Step 5: The latest calories list will be updated to saving file by calling `CalorieList#updateFile()`.
+
+The Class diagram for Calories delete feature is shown below:
+
+![CaloriesDeleteClassDiagram](https://github.com/a-wild-chocolate/tp/blob/master/docs/caloriesDeleteUML.jpg)
```
Comment (cayennegoh) [inline docs/DeveloperGuide.md]: Diagram has formatting issues and does not show up as a picture

File: docs/HydrationDeleteDiagram.png
Comment (cayennegoh) [inline docs/HydrationDeleteDiagram.png:1]: add a end of lifeline "X" after entry deleted

Comment (cayennegoh) [inline docs/HydrationDeleteDiagram.png:1]: Change the # to :
Should be Ui:handleUserInput()

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -5,9 +5,146 @@
...
+
+- Step 4: The hydration record (`Entry`) stored in the `ArrayList<Entry> hydrationList` will be deleted by calling `hydrationArrayList.remove((index-1));` and a successful deleting message will be shown in terminal by calling `HydrationListUi#successfulDeletedMessage(toDelete)`
+
+- Step 5: The latest hydration list will be updated to saving file by calling `HydrationList#updateFile()`.
+
+The Sequence diagram for Hydration delete feature is shown below:
+![HydrationDeleteDiagram.png](HydrationDeleteDiagram.png)
 
 ## Product scope
 ### Target user profile
```
Comment (cayennegoh) [inline docs/DeveloperGuide.md:463]: Could fill in information on scope, user profile etc

File: src/main/java/seedu/lifetrack/system/parser/ParserCalories.java [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+     * delimiters 'desc/', 'c/', 'date/', and 'm/'. The method extracts these components
+     * and creates either an InputEntry or OutputEntry object depending on the user command.
+     * If required inputs are missing or empty, an InvalidInputException is thrown.
+     *
+     * @param input the input string to be parsed, containing date, time, activity,
+     *              and calorie count information
+     * @return an Entry object representing calorie intake
+     * @throws InvalidInputException if the input string is missing components or contains empty fields
+     */
+    public static Entry parseCaloriesInput(String input) throws InvalidInputException {
```
Comment (leongxingyu) [inline src/main/java/seedu/lifetrack/system/parser/ParserCalories.java]: Perhaps can refactor this method more? It seems too wordy.

File: docs/CaloriesAddEntrySeqDiagram.png
Comment (leongxingyu) [inline docs/CaloriesAddEntrySeqDiagram.png:1]: After the updateFile(), from the code it seems like it calls fileHandler in the storage. Perhaps can show the storage class too?

File: docs/HydrationListClassDiagram.png
Comment (leongxingyu) [inline docs/HydrationListClassDiagram.png:1]: Should there be a "C" beside your classes?

Comment (leongxingyu) [inline docs/HydrationListClassDiagram.png:1]: Are the symbols used to represent "private" and "public" methods correct?

File: docs/caloriesDeleteUML.jpg
Comment (leongxingyu) [inline docs/caloriesDeleteUML.jpg:1]: Perhaps can standardise the class diagrams? Some diagrams have colour some doesn't.

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -5,9 +5,147 @@
...
+
+- Step 3: The string will be divided to two substrings according to the command syntax. Index will be tried to get from the second substring by `Integer.parseInt()`.
+
+- Step 4: The calories record (`Entry`) stored in the `ArrayList<Entry> caloriesList` will be deleted by calling `calorieArrayList.remove((index-1));` and a successful deleting message will be shown in terminal by calling `CalorieListUi#successfulDeletedMessage(toDelete)`
+
+- Step 5: The latest calories list will be updated to saving file by calling `CalorieList#updateFile()`.
+
+The Class diagram for Calories delete feature is shown below:
+
+![CaloriesDeleteClassDiagram](https://github.com/a-wild-chocolate/tp/blob/master/docs/caloriesDeleteUML.jpg)
```
Comment (leongxingyu) [inline docs/DeveloperGuide.md]: This diagram does not show. It doesn't seem to appear in the docs package as well.

File: docs/HydrationDeleteDiagram.png
Comment (leongxingyu) [inline docs/HydrationDeleteDiagram.png:1]: This sequence diagram looks different from the previous diagram. Perhaps can standardise it?

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -5,9 +5,147 @@
...
+
+- Step 3: The obtained information is sent to the private method `ParserCalories#makeNewInputEntry(String, int, String)` to create a new entry of class `InputEntry` that extends `Entry`. An `Entry` object is then returned to the caller, `CalorieList#addEntry(String)`.
+
+- Step 4: The returned `Entry` object is added into the `calorieArrayList` member of type `ArrayList<Entry>` in the `CalorieList`, via the `ArrayList.add()` method.
+
+- Step 5: `FileHandler#updateFile()` is then called to update the data file with the new entry in the `CalorieList`.
+
+The sequence diagram for this feature is shown below:
+
+![CaloriesAddEntrySeqDiagram](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/rexyyong/tp/RexDG/docs/CaloriesAddEntrySeqDiagram.puml)
```
Comment (itsmejr257) [inline docs/DeveloperGuide.md]: There appears to be some self-called methods in ParserCalories#parseCaloriesInput() such as checkKeyWordExists(), checkKeywordsCorrectlyOrdered() etc. Would it be better to include these self called methods into the diagram?

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -5,9 +5,147 @@
...
+
+#### Design considerations
+
+- **Alternative 1 (current choice):** Uses an API to get the calories needed
+    - Pros: No need to figure out the optimal algorithm
+    - Cons: Need to parse response to sieve out necessary information
+
+- **Alternative 2:** Uses an algorithm to find the number of calories needed
+  - Pros: Not dependent on external APIs
+  - Cons: Need to come up with an algorithm to use
```
Comment (itsmejr257) [inline docs/DeveloperGuide.md]: Great indication of other design considerations. The implementation for this feature is clearly stated and easy to follow. Would it be better to also include a sequence diagram to trace the code?

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -5,9 +5,147 @@
...
+
+### Calories list feature
+
+The `calories list` feature lists out the record of all the Calories data that the user has keyed in. The Calories data are all stored into a `ArrayList<Entry> calorieArrayList` attribute of the `CalorieList` Class. Calories data are printed when the `printCalorieList()` function is called. 
+
+The `printCalorieList()` function iterates through the `calorieArrayList` and prints out the Entries according to its order in the Array List.
+
+The Class diagram and sequence diagram for Calories list feature is shown below. Unrelated attributes and Classes were excluded.
+
+![CaloriesListClassDiagram](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/rexyyong/tp/DevGuideRex/docs/CaloriesListClassDiagram.puml)
```
Comment (itsmejr257) [inline docs/DeveloperGuide.md]: Good job on using a Class Diagram here. It serves a good purpose for this feature. However, the notation regarding the calories and calorieslist does not seem to be a notation used in CS2113T. Would it be better to use the association lines instead?

File: docs/DeveloperGuide.md (hunk 5 above)
Comment (itsmejr257) [inline docs/DeveloperGuide.md]: The diagram does not seem to appear correctly in the Developer Guide. Would it be better to update the link to a relative path in the code, e.g. "caloriesDeleteUML.jpg" instead of the github link?

File: docs/DeveloperGuide.md [hunk 9]
```diff
@@ -5,9 +5,147 @@
...
+**Ease of Implementation:** Setting a standard hydration requirement simplifies the tracking process for users. It provides a clear goal to strive for, making it easier for individuals to monitor and maintain their hydration levels consistently.
+### Hydration list feature
+
+The `hydration list` feature lists out the record of all the Hydration data that the user has keyed in. The Hydration data are all stored into a `ArrayList<Entry> hydrationArrayList` attribute of the `HydrationList` Class. Hydration data are printed when the `printHydrationList()` function is called.
+
+The `printHydrationList()` function iterates through the `hydrationArrayList` and prints out the Entries according to its order in the Array List.
+
+The Class diagram for Hydration list feature is shown below. Unrelated attributes and Classes were excluded.
+
+![HydrationListClassDiagram.png](HydrationListClassDiagram.png)
```
Comment (itsmejr257) [inline docs/DeveloperGuide.md]: The association between hydration and hydrationlist does not seem to be a notation used in cs2113. Would it be better to update the diagram to use association lines instead?

File: docs/DeveloperGuide.md [hunk 10]
```diff
@@ -5,9 +5,147 @@
...
+- Step 2: After the `Ui#handleHydrationInput()` matching `delete hydration` key word, the string will be passed into deleteEntry(String line) to execute delete process.
+
+- Step 3: The string will be divided to two substrings according to the command syntax. Index will be tried to get from the second substring by `Integer.parseInt()`.
+
+- Step 4: The hydration record (`Entry`) stored in the `ArrayList<Entry> hydrationList` will be deleted by calling `hydrationArrayList.remove((index-1));` and a successful deleting message will be shown in terminal by calling `HydrationListUi#successfulDeletedMessage(toDelete)`
+
+- Step 5: The latest hydration list will be updated to saving file by calling `HydrationList#updateFile()`.
+
+The Sequence diagram for Hydration delete feature is shown below:
+![HydrationDeleteDiagram.png](HydrationDeleteDiagram.png)
```
Comment (itsmejr257) [inline docs/DeveloperGuide.md]: The notation for this sequence diagram appears to be different from the rest of the diagrams. The methods called seem to be indicated with the class. E.g. Ui#handleHydrationinput. Would it be better to standardize the notations to match the previous diagrams, in particular, following the notation taught during lectures?

File: docs/DeveloperGuide.md [hunk 11]
```diff
@@ -5,9 +5,147 @@
...
+
+- Step 2: After the `Ui#handleHydrationInput()` matching `delete hydration` key word, the string will be passed into deleteEntry(String line) to execute delete process.
+
+- Step 3: The string will be divided to two substrings according to the command syntax. Index will be tried to get from the second substring by `Integer.parseInt()`.
+
+- Step 4: The hydration record (`Entry`) stored in the `ArrayList<Entry> hydrationList` will be deleted by calling `hydrationArrayList.remove((index-1));` and a successful deleting message will be shown in terminal by calling `HydrationListUi#successfulDeletedMessage(toDelete)`
+
+- Step 5: The latest hydration list will be updated to saving file by calling `HydrationList#updateFile()`.
+
+The Sequence diagram for Hydration delete feature is shown below:
```
Comment (itsmejr257) [inline docs/DeveloperGuide.md:377]: Great Job at keeping the application used to make the sequence diagrams the same.

**Review by itsmejr257 — COMMENTED** [review]
Some cosmetic changes regarding diagrams used requested. Overall, great job!

File: docs/UserGuide.md [hunk 12]
```diff
@@ -11,9 +32,186 @@
...
+* `calories out Walk to i3 building c/67 d/2022-03-25`
+* `calories out go gym c/300 d/2024-04-03`
+
+### Listing calorie items: `calories list`
+Shows a list of all activities in the calories tracker. Includes both calories in and out.
+
+**Format:**
+`calories list`
+
+#### Sample output
```
Comment (CXIA17) [inline docs/UserGuide.md]: It's good to tell the user about what they can potentially get.

File: docs/UserGuide.md [hunk 13]
```diff
@@ -37,6 +235,17 @@ Example of usage:
 
 ## Command Summary
 
-{Give a 'cheat sheet' of commands here}
-
+| Action                 | Format, Examples                                                           |
```
Comment (CXIA17) [inline docs/UserGuide.md]: Good, a help function can always help the user when they get lost.

File: docs/UserGuide.md [hunk 14]
```diff
@@ -2,7 +2,28 @@
 
 ## Introduction
 
-{Give a product intro}
+LifeTrack is a desktop app for students to track their health data, optimized for use via a Command Line Interface (CLI). It tracks calories, hydration and sleep data for the user, while also providing daily recommendations for calorie and hydration intake, based on the user's build and gender, as well as their body goals and activity levels.
+
+## Quick links
```
Comment (CXIA17) [inline docs/UserGuide.md:7]: A good navigation page

File: docs/DeveloperGuide.md [hunk 15]
```diff
@@ -5,9 +5,147 @@
...
+![HydrationDeleteDiagram.png](HydrationDeleteDiagram.png)
 
 ## Product scope
 ### Target user profile
```
Comment (CXIA17) [inline docs/DeveloperGuide.md:463]: Users have various reasons to choose your product, maybe have more lines and contents for user stories?

File: docs/CaloriesAddEntrySeqDiagram.png
Comment (CXIA17) [inline docs/CaloriesAddEntrySeqDiagram.png:1]: Is it possible to make the bar thicker, it's kinda hard to distinguish the bars when classes are calling multiple functions inside the class.

File: docs/DeveloperGuide.md [hunk 16]
```diff
@@ -5,9 +5,147 @@
...
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+#### Implementation
 
+This functionality is facilitated by `UI`, `CalorieList`, `FileHandler` and `ParserCalories`. It implements one operation, namely:
+- `UI#handleCaloriesInput(String, CalorieList)`
+- `CalorieList#addEntry(String)`
+- `ParserCalories#parseCaloriesInput(String)`
+- `FileHandler#updateFile()`
+
+This feature is activated when the user inputs a `calories in` or `calories out` command in the terminal.
```
Comment (CXIA17) [inline docs/DeveloperGuide.md:171]: Very detailed step descriptions.

File: docs/DeveloperGuide.md [hunk 17]
```diff
@@ -5,9 +5,231 @@
...
+- Step 2: After the `Ui#handleSleepInput()` matching `sleep delete` key word, the string will be passed into deleteSleep(String line) to execute delete process.
+
+- Step 3: The string will be divided to two substrings according to the command syntax. Index will be tried to get from the second substring by `Integer.parseInt()`.
+
+- Step 4: The sleep record (`Entry`) stored in the `ArrayList<Entry> sleepList` will be deleted by calling `sleepArrayList.remove((index-1));` and a successful deleting message will be shown in terminal by calling `SleepListUi#successfulDeletedMessage(toDelete)`
+
+- Step 5: The latest sleep list will be updated to saving file by calling `SleepList#updateFile()`.
+
+The Sequence diagram for Sleep delete feature is shown below:
+![SleepDeleteDiagram.jpg](https://github.com/a-wild-chocolate/tp/blob/f9a94746e763ddf54a4309583b71e0fdbabdb141/docs/SleepDeleteSeqDiagram.jpg)
```
Comment (ShyamKrishna33) [inline docs/DeveloperGuide.md]: The image of the sequence diagram is not rendered correctly. 
Screenshot 2024-04-04 152229

File: docs/DeveloperGuide.md (hunk 9 above)
Comment (ShyamKrishna33) [reply docs/DeveloperGuide.md]: Yeah, this makes sense here.

File: docs/DeveloperGuide.md [hunk 18]
```diff
@@ -5,9 +5,231 @@
...
+
+- Step 3: The obtained information is sent to the private method `ParserCalories#makeNewInputEntry(String, int, String)` to create a new entry of class `InputEntry` that extends `Entry`. An `Entry` object is then returned to the caller, `CalorieList#addEntry(String)`.
+
+- Step 4: The returned `Entry` object is added into the `calorieArrayList` member of type `ArrayList<Entry>` in the `CalorieList`, via the `ArrayList.add()` method.
+
+- Step 5: `FileHandler#updateFile()` is then called to update the data file with the new entry in the `CalorieList`.
+
+The sequence diagram for this feature is shown below:
+
+![CaloriesAddEntrySeqDiagram](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/rexyyong/tp/RexDG/docs/CaloriesAddEntrySeqDiagram.puml)
```
Comment (ShyamKrishna33) [inline docs/DeveloperGuide.md]: Remember to add colon ':' in front of all instances of a class.
Screenshot 2024-04-04 152804

File: docs/DeveloperGuide.md [hunk 19]
```diff
@@ -5,9 +5,231 @@
...
+- Step 5: The latest hydration list will be updated to saving file by calling `HydrationList#updateFile()`.
+
+The Sequence diagram for Hydration delete feature is shown below:
+![HydrationDeleteDiagram.png](HydrationDeleteDiagram.png)
+
+### Adding sleep entries feature
+
+#### Implementation
+
+This functionality is facilitated by `UI`, `SleepList`, `FileHandler` and `ParserSleep`. It implements one operation, namely:
```
Comment (ShyamKrishna33) [inline docs/DeveloperGuide.md:384]: Good job with the implementation!

**Review by ShyamKrishna33 — COMMENTED** [review]
Overall, the DG looks great with minor bugs in rendering sequence diagrams.

File: docs/DeveloperGuide.md [hunk 20]
```diff
@@ -5,9 +5,238 @@
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
```
Comment (nichyjt) [inline docs/DeveloperGuide.md]: This should be at the bottom; and it is not complete

File: docs/DeveloperGuide.md [hunk 21]
```diff
@@ -5,9 +5,238 @@
...
```
Comment (nichyjt) [inline docs/DeveloperGuide.md:142]: Consider adding in a Table of Contents

File: docs/DeveloperGuide.md [hunk 22]
```diff
@@ -5,9 +5,238 @@
...
 
 ## Design & implementation
+### Adding calorie entries feature
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+#### Implementation
```
Comment (nichyjt) [inline docs/DeveloperGuide.md:163]: Good that you zoomed into implementation via the header. What happened to the Design part?

File: docs/DeveloperGuide.md [hunk 23]
```diff
@@ -5,9 +5,238 @@
...
+
+- Step 3: The obtained information is sent to the private method `ParserCalories#makeNewInputEntry(String, int, String)` to create a new entry of class `InputEntry` that extends `Entry`. An `Entry` object is then returned to the caller, `CalorieList#addEntry(String)`.
+
+- Step 4: The returned `Entry` object is added into the `calorieArrayList` member of type `ArrayList<Entry>` in the `CalorieList`, via the `ArrayList.add()` method.
+
+- Step 5: `FileHandler#updateFile()` is then called to update the data file with the new entry in the `CalorieList`.
+
+The sequence diagram for this feature is shown below:
+
+![CaloriesAddEntrySeqDiagram](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/rexyyong/tp/RexDG/docs/CaloriesAddEntrySeqDiagram.puml)
```
Comment (nichyjt) [inline docs/DeveloperGuide.md]: As a peer mentioned, you are missing the colon.
The actor `Bob` is a good adition, but is `Bob` really a good way of describing a `User`?

File: docs/DeveloperGuide.md (hunk 21 above)
Comment (nichyjt) [inline docs/DeveloperGuide.md:142]: For all your images, save it as a proper file inside your docs folder. Avoid copy pasting and letting Github render from blobs. This will help in version control as well.

File: docs/DeveloperGuide.md [hunk 24]
```diff
@@ -5,9 +5,238 @@
...
+
+### Calories list feature
+
+The `calories list` feature lists out the record of all the Calories data that the user has keyed in. The Calories data are all stored into a `ArrayList<Entry> calorieArrayList` attribute of the `CalorieList` Class. Calories data are printed when the `printCalorieList()` function is called. 
+
+The `printCalorieList()` function iterates through the `calorieArrayList` and prints out the Entries according to its order in the Array List.
+
+The Class diagram and sequence diagram for Calories list feature is shown below. Unrelated attributes and Classes were excluded.
+
+![CaloriesListClassDiagram](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/rexyyong/tp/DevGuideRex/docs/CaloriesListClassDiagram.puml)
```
Comment (nichyjt) [inline docs/DeveloperGuide.md]: The class diagram notation has issues as your peers pointed out.
Remove the (C) icon and fix the access modifiers notation.

File: docs/DeveloperGuide.md [hunk 25]
```diff
@@ -5,9 +5,238 @@
...
+![CaloriesListSequenceDiagram](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/rexyyong/tp/RexDG/docs/CaloriesListSequenceDiagram.puml)
```
Comment (nichyjt) [inline docs/DeveloperGuide.md]: A few points to note:

1. Im not sure about what you are trying to convey in this diagram. The `get` activation bar seems to extend beyond the `loop` frame, which implies that the act of `get`ting breaks out of the loop?

image

2. Missing colon for sequence diagram

File: docs/DeveloperGuide.md [hunk 26]
```diff
@@ -5,9 +5,238 @@
...
+
+- Step 3: The string will be divided to two substrings according to the command syntax. Index will be tried to get from the second substring by `Integer.parseInt()`.
+
+- Step 4: The calories record (`Entry`) stored in the `ArrayList<Entry> caloriesList` will be deleted by calling `calorieArrayList.remove((index-1));` and a successful deleting message will be shown in terminal by calling `CalorieListUi#successfulDeletedMessage(toDelete)`
+
+- Step 5: The latest calories list will be updated to saving file by calling `CalorieList#updateFile()`.
+
+The Class diagram for Calories delete feature is shown below:
+
+![CaloriesDeleteClassDiagram](https://github.com/a-wild-chocolate/tp/blob/master/docs/caloriesDeleteUML.jpg)
```
Comment (nichyjt) [inline docs/DeveloperGuide.md]: Standardize what tools you all use for Class and UML diagrams. Multiple syntax issues here: the diagram notation is not as per what the course has taught.

For example, your association labels should not be `<<label>>`.

File: docs/DeveloperGuide.md [hunk 27]
```diff
@@ -5,9 +5,238 @@
...
+
+- Step 3: The created `HydrationEntry` instance is added into the `ArrayList<Entry>` attribute of the `HydrationList`.
+
+### Calculating hydration requirements for each user
+
+#### Design Considerations
+
+**General Health Guidelines:** The recommended daily intake of water for an average adult is around 8 glasses or approximately 2000 milliliters. This guideline is commonly recommended by health authorities and organizations such as HealthHub.
+
+**Ease of Implementation:** Setting a standard hydration requirement simplifies the tracking process for users. It provides a clear goal to strive for, making it easier for individuals to monitor and maintain their hydration levels consistently.
```
Comment (nichyjt) [inline docs/DeveloperGuide.md]: Excellent!

File: docs/DeveloperGuide.md [hunk 28]
```diff
@@ -5,9 +5,238 @@
...
+- Step 2: After the `Ui#handleHydrationInput()` matching `delete hydration` key word, the string will be passed into deleteEntry(String line) to execute delete process.
+
+- Step 3: The string will be divided to two substrings according to the command syntax. Index will be tried to get from the second substring by `Integer.parseInt()`.
+
+- Step 4: The hydration record (`Entry`) stored in the `ArrayList<Entry> hydrationList` will be deleted by calling `hydrationArrayList.remove((index-1));` and a successful deleting message will be shown in terminal by calling `HydrationListUi#successfulDeletedMessage(toDelete)`
+
+- Step 5: The latest hydration list will be updated to saving file by calling `HydrationList#updateFile()`.
+
+The Sequence diagram for Hydration delete feature is shown below:
+![HydrationDeleteDiagram.png](HydrationDeleteDiagram.png)
```
Comment (nichyjt) [inline docs/DeveloperGuide.md]: Same as `itsmejr257` pointed out

File: docs/DeveloperGuide.md [hunk 29]
```diff
@@ -5,9 +5,238 @@
...
+- Step 2: Inside `SleepList#addSleep(String)`, the function `ParserSleep#parseSleepInput(String)` is then called to extract information such as the duration and date of entry.
+
+- Step 3: It will create a new entry of class `SleepEntry` that extends `Entry`based on the information.
+
+- Step 4: The created `SleepEntry` instance is added into the `ArrayList<Entry> sleepList` attribute of the `SleepList`.
+
+- Step 5: `FileHandler#updateFile()` is then called to update the data file with the new entry in the `SleepList`.
+
+The sequence diagram for this feature is shown below:
+![SleepAddSeqDiagram](https://github.com/a-wild-chocolate/tp/blob/f9a94746e763ddf54a4309583b71e0fdbabdb141/docs/SleepAddSeqDiagram.jpg)
```
Comment (nichyjt) [inline docs/DeveloperGuide.md]: Standardize the tools used for the diagrams.
A few issues: 

1. No return from activation bar

image

2. Return line not starting at end of activation bar. 
image

3. User does not need an activation bar

File: docs/DeveloperGuide.md [hunk 30]
```diff
@@ -5,9 +5,238 @@
...
+
+
+### Sleep list feature
+
+The `sleep list` feature lists out the record of all the sleep data that the user has keyed in. The sleep data are all stored into a `ArrayList<Entry> sleepList` attribute of the `SleepList` Class. Sleep data are printed when the `printSleepList()` function is called.
+
+The `printSleepList()` function iterates through the `sleepList` and each entry will call `SleepEntry#toString()` to return its information string to be printed.
+
+The Sequence diagram for Sleep list feature is shown below. Unrelated attributes and Classes were excluded.
+![SleepListSeqDiagram](https://github.com/a-wild-chocolate/tp/blob/f9a94746e763ddf54a4309583b71e0fdbabdb141/docs/SleepListSeqDiagram.jpg)
```
Comment (nichyjt) [inline docs/DeveloperGuide.md]: Same comments as per the previous seq diagram critique

File: docs/DeveloperGuide.md [hunk 31]
```diff
@@ -5,9 +5,238 @@
...
+- Step 3: The string will be divided to two substrings according to the command syntax. Index will be tried to get from the second substring by `Integer.parseInt()`.
+
+- Step 4: The sleep record (`Entry`) stored in the `ArrayList<Entry> sleepList` will be deleted by calling `sleepArrayList.remove((index-1));` and a successful deleting message will be shown in terminal by calling `SleepListUi#successfulDeletedMessage(toDelete)`
+
+- Step 5: The latest sleep list will be updated to saving file by calling `SleepList#updateFile()`.
+
+The Sequence diagram for Sleep delete feature is shown below:
+![SleepDeleteDiagram.jpg](https://github.com/a-wild-chocolate/tp/blob/f9a94746e763ddf54a4309583b71e0fdbabdb141/docs/SleepDeleteSeqDiagram.jpg)
+
+### Calculating sleep requirements for each user (Planning)
```
Comment (nichyjt) [inline docs/DeveloperGuide.md:454]: May want to rephrase as proposed future feature

File: docs/DeveloperGuide.md [hunk 32]
```diff
@@ -5,9 +5,238 @@
...
+
+#### Design Considerations
+
+**General Health Guidelines:** The recommended daily sleep duration for an average adult is around 7.5 hours. However, sleep standard is different for different individual with different healthy status. This feature will calculate recommend sleep duration by formula based on user health information.
+
+**Ease of Implementation:** Setting a standard sleep requirement simplifies the tracking process for users. It provides a clear goal to strive for, making it easier for individuals to monitor and maintain their sleep time levels consistently.
 
 ## Product scope
 ### Target user profile
```
Comment (nichyjt) [inline docs/DeveloperGuide.md]: Missing content

**Review by nichyjt — COMMENTED** [review]
Hi team,

Good job at making the DG!  

Take note of these general comments:
1. Please take note of the syntax issues with the diagrams. **Standardize your diagram tools**.
2. There are some sections of the DG that are incomplete. Please finish them.
3. The parts where design considerations were added is good. Don't forget to include design considerations and the design of the feature itself (class diagram) as sections.
4. Upload your diagrams as files, do not use Github blobs (copy paste directly into the Github code editor).

You may refer to the [AB3 DG](https://se-education.org/addressbook-level3/DeveloperGuide.html) as reference on how a well structured DG looks like as you fix your DG and add more diagrams/content.
